[danplatypus30] iP

danplatypus30 opened the pull request:
# Miku
> "Ohayo Oni Chan!!! ~~" ~ Miku ([source](www.google.com))

Miku frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](www.google.com).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE!**

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

danplatypus30 committed:
changed duke to miku, base done

danplatypus30 committed:
removed duke

danplatypus30 committed:
added echo functions

danplatypus30 committed:
added list features, all input is added to list, list is flushed after listed

danplatypus30 committed:
removed flushing of list

danplatypus30 committed:
added Task class, added mark features and unmark features

danplatypus30 committed:
added three new classes, subclasses of Task

danplatypus30 committed:
working todo, deadline, events, with mark and unmark

danplatypus30 committed:
added try catch for mark and unmark as well

danplatypus30 committed:
set up basic test case for most inputs, including mark, unmark, todo, deadline, event

danplatypus30 committed:
added custom exception EmptyStringException

danplatypus30 committed:
added delete function

fishnotphish30 committed:
fix docs: Duke -> Miku

danplatypus30 committed:
Merge pull request #1 from fishnotphish30/fix-docs

fix docs: Duke -> Miku

danplatypus30 committed:
added save function

danplatypus30 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  added save function

danplatypus30 committed:
added date time compatibility

danplatypus30 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  added date time compatibility

danplatypus30 committed:
changed all to OOP format

danplatypus30 committed:
Merge branch 'master' of https://github.com/danplatypus30/ip into branch-Level-8

danplatypus30 committed:
put all classes into duke package

danplatypus30 committed:
Merge branch 'add-gradle-support'

danplatypus30 committed:
can build and run using gradle, ./gradlew <commands>

danplatypus30 committed:
basic JUnit test added

danplatypus30 committed:
more JUnit tests added, fileIO

danplatypus30 committed:
Added JavaDocs comments for all classes and methods

danplatypus30 committed:
checkstyle passed now

danplatypus30 committed:
added find function

danplatypus30 committed:
resolved merge conflicts

danplatypus30 committed:
checkstyle issues resolved

yuyang011 commented on src/main/java/duke/Miku.java:
```diff
@@ -0,0 +1,53 @@
+ * Represents the main class of the program.
+ */
+public class Miku {
+    private UI ui;
```
Should UI be Ui instead to follow the PascalCase naming convention for a class name?

yuyang011 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+            }
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("unable to update file, error...");
```
Should this exception be printing instead of throwing an error?

yuyang011 commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,202 @@
+                    LocalDateTime dt = Task.convert(parts[1].trim());
+                    LocalDateTime dt2 = Task.convert(parts[2].trim());
+                    Events ee = new Events(parts[0].trim(), dt, dt2);
+                    list = list.add(ee);
```
Should the use of ee to name the event be replaced with a better name to avoid confusion to readers, as it is used in another subsequent line?

yuyang011 commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,202 @@
+                try {
+                    String[] parts = userIn.substring(9).split("/by");
+                    LocalDateTime dt = Task.convert(parts[1].trim());
+                    Deadlines dd = new Deadlines(parts[0].trim(), dt);
```
Should the use of dd to name the event be replaced with a better name to avoid confusion to readers, as it is used in another subsequent line?

yuyang011 commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,202 @@
+                        throw new EmptyStringException("String cannot be empty");
+                    }
+                    ToDos td = new ToDos(userIn.substring(5));
+                    list = list.add(td);
```
Should the use of td to name the event be replaced with a better name to avoid confusion to readers, as it is used in another subsequent line?

yuyang011 reviewed (commented):
LGTM, just some minor coding standards to fix

JadeCheah commented on src/main/java/duke/Miku.java:
```diff
@@ -0,0 +1,53 @@
+package duke;
```
Maybe consider renaming the package to miku for consistency? since the chatbot is named Miku but the package is duke.

JadeCheah commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,155 @@
+     * @param datestring
+     * @return
+     */
+    public static LocalDateTime convert(String datestring) {
```
Perhaps can use a more descriptive name like convertTo (since the other one is convertBack) or maybe convertToDateObject?

JadeCheah commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+        File file = new File(filepath);
+
+        // create file if file does not exist
+        if (file.exists() == false) {
```
Would this be a better alternative?
```suggestion
        if (!file.exists()) {
```

JadeCheah commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,202 @@
+    /**
+     * Runs a while loop that interacts with the user.
+     */
+    public void interact() {
```
interact() function seems to be handling both the parsing logic and the user interaction, which could be too long and a bit hard to maintain. Maybe can consider creating a Parser class to handle command extraction and argument processing?

JadeCheah commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,88 @@
+/**
+ * Represents a deadline task.
+ */
+public class Deadlines extends Task {
```
Maybe Deadline (without the s) would be a more suitable name for this class? Same applies to Events, ToDos.

JadeCheah reviewed (commented):
Overall well structured and easily understandable. A minor suggestion is probably try to keep similar classes in one package (e.g. task package for Deadline, Event, Task and ToDo). Other aspects lgtm, keep up the good work!

danplatypus30 committed:
migrated the whole app to GUI

danplatypus30 committed:
Update Deadlines.java

danplatypus30 committed:
Update DialogBox.java

danplatypus30 committed:
Update DialogBox.java again

danplatypus30 committed:
Update DialogBox.java again

danplatypus30 committed:
Update Launcher.java

danplatypus30 committed:
Merge branch 'master' of https://github.com/danplatypus30/ip

* 'master' of https://github.com/danplatypus30/ip:
  Update Launcher.java
  Update DialogBox.java again
  Update DialogBox.java again
  Update DialogBox.java
  Update Deadlines.java
  migrated the whole app to GUI

danplatypus30 committed:
Change exceptions to be more accurate

danplatypus30 committed:
Bug fixes for Storage.java

Some errors happened with ./gradlew run previously because Storage.java has a try catch block that catches UnexpectedException.

But the try catch block will never throw an UnexpectedException so I removed it.

danplatypus30 committed:
Add assert functions

Added some assert statements in Duke.java.

Added enableassertion flag in run function in build.gradle.

danplatypus30 committed:
Change many classes for refactoring

Major refactoring works especially to shorten the long decision method in Duke.java.

danplatypus30 committed:
Merge pull request #3 from danplatypus30/branch-A-Assertions

Add assert functions

danplatypus30 committed:
Revert "Add assert functions"

danplatypus30 committed:
Merge pull request #5 from danplatypus30/revert-3-branch-A-Assertions

Revert "Add assert functions"

danplatypus30 committed:
Merge branch 'master' of https://github.com/danplatypus30/ip

* 'master' of https://github.com/danplatypus30/ip:
  Revert "Add assert functions"
  Add assert functions

danplatypus30 committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Revert "Add assert functions"
  Add assert functions

danplatypus30 committed:
Merge pull request #4 from danplatypus30/branch-A-CodeQuality

Change many classes for refactoring

danplatypus30 committed:
Merge branch 'master' of https://github.com/danplatypus30/ip

* 'master' of https://github.com/danplatypus30/ip:
  Change many classes for refactoring

danplatypus30 committed:
Add new codes to track expenses

Can add expenses for each task using /expenses <double>

Deleting tasks will reduce the expenses automatically

danplatypus30 committed:
Merge branch 'D-Expenses'

* D-Expenses:
  Add new codes to track expenses

danplatypus30 committed:
Add User Guide

danplatypus30 committed:
Update README.md for github pages

danplatypus30 committed:
Add more JUnit tests and update docs

danplatypus30 committed:
Clean Dates and Title Bar

Realised that my date input and output are the same format, which does not meet the requirements. Updated it.

Also added some text for my title bar because its empty.

danplatypus30 committed:
Fix Fatal DateTime Conversion Bug

Also fixed multiple checkstyle errors
